var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -431,8 +430,9 @@ def cmake_args(self):
         if self.spec.satisfies("@2020:"):
             options.append("-DGMX_INSTALL_LEGACY_API=ON")
 
-        if self.spec.satisfies("%oneapi"):
-            options.append("-DGMX_GPLUSPLUS_PATH=%s/g++" % self.spec["gcc"].prefix.bin)
+        if self.spec.satisfies("%oneapi") or self.spec.satisfies("%intel"):
+            with open(".".join([os.environ["SPACK_CXX"], "cfg"]), "r") as f:
```

Comment:
This might throw an error when the file does not exist; this cfg file is not standard correct?

Reply:
It's written by `spack install intel-oneapi-compilers` here https://github.com/spack/spack/blob/1af7284b5d3958414edff83cd7104cffbbe4c125/var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py#L255

So, yes. If compiler is installed outside of Spack this file might not exist.

Reply:
So, I can use the old installation as fallback in case the file does not exists, but how to I get the conditional `depends_on("gcc")`? This fails as `SPACK_CXX` is not defined outside of any function in`package.py`:
```
    if not os.path.exists(".".join([os.environ["SPACK_CXX"], "cfg"])):
        depends_on("gcc", when="%oneapi")
        depends_on("gcc", when="%intel")
```

Reply:
OK. @becker33 suggested to use a variant.